crates/boundless-market/src/selector.rs

```diff
@@ -15,16 +15,120 @@
 //! Selector utility functions.
 
 use std::collections::HashMap;
+use std::fmt::{self, Display, Formatter};
 
 use alloy_primitives::FixedBytes;
 use clap::ValueEnum;
 use risc0_aggregation::SetInclusionReceiptVerifierParameters;
-use risc0_ethereum_contracts::selector::{Selector, SelectorType};
+use risc0_ethereum_contracts::selector::Selector;
 use risc0_zkvm::sha::{Digest, Digestible};
+use thiserror::Error;
 
 use crate::contracts::UNSPECIFIED_SELECTOR;
 use crate::util::is_dev_mode;
 
+#[derive(Debug, Error)]
+#[non_exhaustive]
+/// Errors related to extended selectors.
+pub enum SelectorExtError {
+    /// Unsupported selector error.
+    #[error("Unsupported selector")]
+    UnsupportedSelector,
+    /// No verifier parameters error.
+    #[error("Selector {0} does not have verifier parameters")]
+    NoVerifierParameters(SelectorExt),
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[non_exhaustive]
+/// Types of extended selectors.
+pub enum SelectorExtType {
+    /// A fake receipt selector used for testing and development.
+    FakeReceipt,
+    /// Groth16 proof selector.
+    Groth16,
+    /// Set verifier selector.
+    SetVerifier,
+    /// Blake3 Groth16 selector.
+    Blake3Groth16,
+}
+
+#[repr(u32)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[non_exhaustive]
+/// Extended selectors for various proof types.
+pub enum SelectorExt {
+    /// A fake receipt selector used for testing and development.
+    FakeReceipt = 0xFFFFFFFF,
+    /// Groth16 proof selector version 3.0.
+    Groth16V3_0 = 0x73c457ba,
```

Comment:
Instead of hard coding it, should we reimport from r0-ethereum for the selectors that they manage?